### PR TITLE
manual cloud stuff

### LINKED ISSUE #971: clouds-manual page needs rewriting
we need to have a new manual provisioning page for 2.0

### PR DESCRIPTION
**Pull request by evilnick**
Fixes #971

### CODE REVIEW

File: src/en/clouds-manual.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  - You will need at least two machines
+  - The machines must already have Ubuntu installed
+  - They must be running SSH
+  - You will need a login with sudo privileges on all machines
+  - All machines must be able to communicate with each other over the network
+
+## Bootstrapping the cloud
+
+As with other clouds, Juju needs to create a controller to manage models and 
+them other instances in the cloud. Bootstrapping is the same as for 'regular'
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: them other instances?

File: src/en/clouds-manual.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+As with other clouds, Juju needs to create a controller to manage models and 
+them other instances in the cloud. Bootstrapping is the same as for 'regular' 
+clouds, except in this case, after the cloud type you should supply the network 
+address of the machine you wish to use as the controller:
+
+```bash
+juju bootstrap mycloud manual/192.168.1.128
+```
+
+Note that it is also possible to use the other common bootstrap parameters here.
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: maybe provide a link

File: src/en/clouds-manual.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
+Title: Adding a manual cloud
+
+# Using a "Manual" cloud
+
+As well as many other types of cloud, public and private, Juju caters for the
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: this beginning is awkward

File: src/en/clouds-manual.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+```bash
+juju bootstrap mycloud manual/192.168.1.128
+```
+
+Note that it is also possible to use the other common bootstrap parameters here.
+
+## Adding machines to the cloud
+
+Bootstrapping the cloud created the controller and initial models that Juju
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: creates?

File: src/en/clouds-manual.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+though. The more conventional Juju cloud types can create machines
+automatically, but in the case of a manual cloud you will have to specify these
+resources.
+
+This is done by using the `add-machine` command in Juju. This will install the 
+default series on a given resource (in this case a machine on the network to
+which you have access and the relevant permissions), and prepare it for
+deploying services.
+
+For example, if you have a machine at the IP address 192.168.1.129, you could
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: machine with IP address?

File: src/en/clouds-manual.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+Bootstrapping the cloud created the controller and initial models that Juju
+uses. Deploying services will require additional machines to be made available
+though. The more conventional Juju cloud types can create machines
+automatically, but in the case of a manual cloud you will have to specify these
+resources.
+
+This is done by using the `add-machine` command in Juju. This will install the 
+default series on a given resource (in this case a machine on the network to
+which you have access and the relevant permissions), and prepare it for
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: i wonder if we need to repeat the requirements, it was mentioned in the beginning

File: src/en/clouds-manual.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+deploying services.
+
+For example, if you have a machine at the IP address 192.168.1.129, you could 
+add it to your manual cloud with the command:
+
+```bash
+juju add-machine ssh:ubuntu@192.168.1.129
+```
+
+In this case, assuming SSH is available and there is a user called 'ubuntu' on
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: i wonder if we need to repeat the requirements, it was mentioned in the beginning

File: src/en/clouds-manual.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+that machine, you will get a password request. As the preparation requires
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: s/request/prompt ?

File: src/en/clouds-manual.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+sudo privileges on the remote machine, you will be asked for the password once
+again as the installer initialises.
+
+The process should not take too long, and you will see messages on the screen as
+the various stages take place. Once the command has returned, you can check
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: stages do not take place. activities in the stages take place. consider rewording somehow

File: src/en/clouds-manual.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+Usually Juju will create machines as it needs them. By default, using a charm
+to deploy a service will automatically create a machine first and then 
+deploy the new service. 
+
+With a manual cloud, you need to create your machines in advance, so Juju will
+need to know which of the machines you wish to target when you deploy a charm
+(or scale out). This can be done with the use of the placement directives.
+Deploying with placement is covered in the 
+[documentation for `deploy`][placement], but essentially, you need to use the
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: consider making the first occurrence of 'placement directives' the link and trashing 'Deploying with...' and continue with 'You need to use'

File: src/en/clouds-manual.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+[documentation for `deploy`][placement], but essentially, you need to use the 
+'--to' option to specify the destination, which is either a machine or a 
+container on a machine:
+
+```bash
+juju deploy wordpress --to 0
+```
+...will deploy WordPress on the first machine you added.
+
+## Additional Notes
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: Additional notes

File: src/en/clouds-manual.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+There are some additional things to note when using the Manual provider:
+
+ - Machines are known to the model Juju is using. If you destroy the model and 
+   create a new one, you will have to add the machines again to the new model.
+ - To improve performance, consider running a local apt proxy (see also
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: APT

File: src/en/clouds-manual.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+
+# Using a "Manual" cloud
+
+As well as many other types of cloud, public and private, Juju caters for the 
+case where you may not be able to access one of these traditional clouds in a 
+straightforward way; maybe you can't create additional instances, or perhaps 
+your cloud is really a collection of disparate hardware.
+
+Whatever the case, as long as Juju can log into these machines (and you have 
+sudo permissions on them) they can be used as part of a Juju cloud. It won't
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: you repeat the requirements below. consider removing here

File: src/en/clouds-manual.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+be _quite_ the same as using a standard cloud - without the ability to create
+new instances when they are desired, you will be missing out on some of the 
+Juju magic. You can still deploy and manage services thoguh, with a bit of extra
```
Comment (pmatulis) [inline src/en/clouds-manual.md]: what is this magic you speak of stranger? :fearful:

**Comment by pmatulis** [comment]
Review complete.

File: src/en/clouds-manual.md (hunk 14 above)
Comment (evilnick) [inline src/en/clouds-manual.md]: too spooky for you

File: src/en/clouds-manual.md (hunk 7 above)
Comment (evilnick) [inline src/en/clouds-manual.md]: We never mentioned a user called 'ubuntu' before. i think it is fine here.

**Pull request merged**